apisix/error_handler.lua

```diff
@@ -0,0 +1,53 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local ngx = ngx
+
+
+local html_5xx = [[
+<!DOCTYPE html>
+<html>
+<head>
+<meta content="text/html;charset=utf-8" http-equiv="Content-Type">
+<meta content="utf-8" http-equiv="encoding">
+<title>Error</title>
+<style>
+    body {
+        width: 35em;
+        margin: 0 auto;
+        font-family: Tahoma, Verdana, Arial, sans-serif;
+    }
+</style>
+</head>
+<body>
+<h1>An error occurred.</h1>
+<p>Sorry, the page you are looking for is currently unavailable.<br/>
+Please try again later.</p>
+<p>If you are the system administrator of this resource then you should check
+the <a href="https://github.com/apache/apisix/blob/master/conf/config-default.yaml#L135">error log
+</a> for details.</p>
+<p>If you need any help, click <a href="https://apisix.apache.org/">need help</a>.</em></p>
+<p><em>Faithfully yours, <a href="https://apisix.apache.org/">APISIX</a>.</em></p>
+</body>
+</html>
+]]
+
+
+return function()
```

Comment:
this is not a good way, we mainly use `Lua module` way.

it is easy to add new features later.

Reply:
What's the `lua module` way?
Do you mean we can create a module like error_handler? Then we can have

```
_M.5xx_error 
_M.4xx_error
```

Reply:
Here is an example:

```lua
local _M = {}

local function 5xx_error()
    ... 
end 

local function 4xx_error()
    ... 
end 

function _M.error_page(status)
    if ... then
        return 5xx_error()
    end

    if ... then
        return 4xx_error()
    end

    ... ...
end

return _M
```